gfx/common/x11_common.c

```diff
@@ -147,6 +153,75 @@ void x11_set_window_attr(Display *dpy, Window win)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
+    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+#ifdef HAVE_DBUS
+    x11_suspend_screensaver_dbus(enable);
+#endif
+}
+
+#ifdef HAVE_DBUS
+void x11_suspend_screensaver_dbus(bool enable)
+{
+    const char *app = "RetroArch";
+    const char *reason = "Playing a game";
+    DBusMessage *msg, *reply;
+
+    if (!enable)
```

Comment:
Then that too is buggy. No point fixing it in this PR, but let's not clone the same mistake elsewhere. You've got an uninhibit function in x11_window_destroy, just move it here. Minus the dbus disconnect stuff, there's no need for that at all; the kernel will disconnect you when the process exits.